ReactWindows/ReactNative/UIManager/FutureAccessList.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Threading.Tasks;
+using Windows.Storage;
+using Windows.Storage.AccessCache;
+
+namespace ReactNative.UIManager
+{
+    /// <summary>
+    /// The reason this future access list thing was introduced was a need
+    /// to transfer a file object from RN to JS and then to another RN module.
+    /// The naive approach to transfer the file path didn't work because the
+    /// file path doesn't carry read/write permissions and a UWP app can directly
+    /// access only files inside its app folder.
+    ///
+    /// Eventually, RNW will have a more general approach to store such things
+    /// and it will support other schemes like blob. At that time this class will
+    /// likely to be rewritten.
+    /// </summary>
+    static class FutureAccessList
+    {
+        private const string PREFIX = "urn:future-access-list:";
+
+        /// <summary>
+        /// Stores a file object in the future access list cache.
+        /// Later the file object can be retrieved with the Get(token) method.
+        /// The capacity of the cache is limited (up to about 1000 items) and
+        /// if necessary old items can be removed to free space for new ones.
+        /// </summary>
+        public static string Add(StorageFile file)
+        {
+            var token = GetAccessToken(file);
+            return PREFIX + Uri.EscapeUriString(token);
+        }
+
+        /// <summary>
+        /// Resolves a URI returned by the Add(file) method to the file object.
+        /// The file object remains in the cache and can be retrieved again later.
+        /// </summary>
+        public static async Task<StorageFile> Get(string uri)
```

Comment:
>Get [](start = 46, length = 3)

Let's say I want to display an image that was dropped and added to the FAL. Do I need to refer to the FAL from ImageLoaderModule to use it?

Doesn't have to be done here, but I think we'll need a more general solution for URL resolution (e.g., for blob module #1471)

Reply:
Just realized you already discussed adding a content provider.

---
In reply to: [174758293](https://github.com/Microsoft/react-native-windows/pull/1549#discussion_r174758293) [](ancestors = 174758293)

Reply:
That's right, `ImageLoaderModule` will need to refer to the FAL.